src/WSRequestHandler_ComponantsManagement.cpp

```
@@ -0,0 +1,128 @@
+{
+	if (!req->hasField("sourceName") || !req->hasField("typeIDName") || !req->hasField("sourceSettings")) {
+		return req->SendErrorResponse("missing request parameters");
+	}
+	const char* sceneName = obs_data_get_string(req->data, "sceneName");
+	obs_scene_create(sceneName);
+	OBSDataAutoRelease response = obs_data_create();
+	obs_data_set_string(response, "sceneName", sceneName);
+	return req->SendOKResponse(response);
+}
+
+/**
+* Add Active Child
+*
+* @param {String} `sourceParentName` Name of the parent source
+* @param {String} `sourceChildName` Name of the child source
+*
+* @api requests
+* @name AddActiveChild
+* @category sources
+*/
+HandlerResponse WSRequestHandler::HandleAddActiveChild(WSRequestHandler* req)
+{
+	if (!req->hasField("sourceParentName") || !req->hasField("sourceChildName")) {
+		return req->SendErrorResponse("missing request parameters");
+	}
+	const char* sourceParentName = obs_data_get_string(req->data, "sourceParentName");
+	const char* sourceChildName = obs_data_get_string(req->data, "sourceChildName");
+	obs_source_t* sourceParent= obs_get_source_by_name(sourceParentName);
+	obs_source_t* sourceChild = obs_get_source_by_name(sourceChildName);
+	
+	obs_source_add_active_child(sourceParent, sourceChild);
+	return req->SendOKResponse();
+}
+
+/**
+* Add source to current Scene 
+*
+* @param {String} `sourceName` Name of the source to be added
+*
+* @api requests
+* @name AddtoScene 
+* @category sources
+*/
+HandlerResponse WSRequestHandler::HandleAddtoScene(WSRequestHandler* req)
+{
+	if (!req->hasField("sourceName")) {
+		return req->SendErrorResponse("missing request parameters");
+	}
+
+	const char* targetSource = obs_data_get_string(req->data, "sourceName");
+
+	obs_source_t *Source= obs_get_source_by_name(targetSource);
+	obs_source_t *Scenetarget = obs_frontend_get_current_preview_scene();
+	obs_scene_t  *Scene =obs_scene_from_source(Scenetarget);
+	obs_scene_add(Scene, Source);
```

Please return an error if this call returns a null pointer.